src/modules/Channel/components/MessageList/index.tsx

```diff
@@ -245,29 +273,34 @@ const MessageList: React.FC<MessageListProps> = ({
         {currentGroupChannel?.isFrozen && (
           <FrozenNotification className="sendbird-conversation__messages__notification" />
         )}
-        {(unreadSince || unreadSinceDate) && (
-          <UnreadCount
-            className="sendbird-conversation__messages__notification"
-            count={currentGroupChannel?.unreadMessageCount}
-            time={unreadSince}
-            lastReadAt={unreadSinceDate}
-            onClick={() => {
-              if (scrollRef?.current?.scrollTop) {
-                scrollRef.current.scrollTop = (scrollRef?.current?.scrollHeight ?? 0) - (scrollRef?.current?.offsetHeight ?? 0);
-              }
-              if (!disableMarkAsRead && !!currentGroupChannel) {
-                markAsReadScheduler.push(currentGroupChannel);
-                messagesDispatcher({
-                  type: messageActionTypes.MARK_AS_READ,
-                  payload: { channel: currentGroupChannel },
-                });
-              }
-              setInitialTimeStamp(null);
-              setAnimatedMessageId(null);
-              setHighLightedMessageId(null);
-            }}
-          />
-        )}
+        {
+          /**
+           * Show unread count IFF scroll is not bottom or is bottom but hasNext is true.
+           */
+          (!isScrollBottom || hasMoreNext) && (unreadSince || unreadSinceDate) && (
+            <UnreadCount
+              className="sendbird-conversation__messages__notification"
+              count={currentGroupChannel?.unreadMessageCount}
+              time={unreadSince}
+              lastReadAt={unreadSinceDate}
+              onClick={() => {
+                if (scrollRef?.current?.scrollTop) {
+                  scrollRef.current.scrollTop = (scrollRef?.current?.scrollHeight ?? 0) - (scrollRef?.current?.offsetHeight ?? 0);
+                }
```

Comment:
Is this logic for scrolling to the bottom?
```suggestion
                if (scrollRef?.current) {
                  scrollRef.current.scrollTop = Number.MAX_SAFE_INTEGER
                }
```

Reply:
yeah it seems like it is hitting the bottom most. At first i thought this is going to top of the last message but when i tested it was going to the bottom of the last message. So i think we can change to your suggested.